File: LV1/1.5/ptice/ptice.rb
_ = gets
exam = gets.chomp.chars

a = ('ABC' * 40).chars
b = ('BABC' * 25).chars
g = ('CCAABB' * 20).chars
score_pack = a.zip(b, g)
score = [0, 0, 0]

exam.zip(score_pack) do |pack|
  res = pack[1].map { |p| p == pack[0] && 1 || 0 }
  score = score.zip(res).map(&:sum)
end

name = ['Adrian', 'Bruno', 'Goran']
max_score = score.max

puts max_score
ans = score.zip(name).select { |sc, *| sc == max_score }.map { |*, na| na }
puts ans.join("\n")

File: LV1/1.5/ptice/ptice.py
_ = int(input())
exam = input()

a = 'ABC'*40
b = 'BABC'*25
g = 'CCAABB'*20
score_pack = zip(a, b, g)
score = [0, 0, 0]

for c in exam:
    res = [c == ans for ans in next(score_pack)]
    score = [a+b for (a, b) in zip(score, res)]

name = ["Adrian", "Bruno", "Goran"]
max_score = max(score)

print(max_score)
print("\n".join([ni for (ni, sr) in zip(name, score) if sr == max_score]))
